src/vsr/replica.zig

```diff
@@ -4687,12 +4925,12 @@ pub fn ReplicaType(
             assert(self.primary());
 
             assert(message.header.command == .request);
-            assert(message.header.client > 0);
             assert(message.header.view <= self.view); // See ignore_request_message_backup().
 
             if (self.pipeline.queue.message_by_client(message.header.client)) |pipeline_message| {
                 assert(pipeline_message.header.command == .request or
                     pipeline_message.header.command == .prepare);
+                assert(message.header.client != 0);
```

Comment:
Since `client=0` has no session, the `pipeline.queue.message_by_client` case is already covered for client 0 in `ignore_request_message_duplicate()`.